spec/turn_message_spec.rb

```diff
@@ -0,0 +1,126 @@
+          expect(described_class.for(turn_message)).to eq("Two correct guess at the exact position")
+        end
+      end
+
+      context "three colors guessed at the exact position" do
+        it "returns a message for three correct guess colors" do
+          turn = Turn.new(passcode: passcode)
+          turn_message = turn.guess(["RED", "GREEN", "BLUE", "ORANGE"])
+          expect(described_class.for(turn_message)).to eq("Three correct guess at the exact position")
+        end
+      end
+    end
+
+    describe "partial position" do
+      context "all colors guessed at the wrong position" do
+        it "returns a message for four partial guess colors" do
+          turn = Turn.new(passcode: passcode)
+          turn_message = turn.guess(["YELLOW", "BLUE", "GREEN", "RED"])
+          expect(described_class.for(turn_message)).to eq("Four colors guessed at the wrong position")
+        end
+      end
+
+      context "three colors guessed at the wrong position" do
+        it"returns a message for three partial guess colors" do
+          turn = Turn.new(passcode: passcode)
+          turn_message = turn.guess(["ORANGE", "BLUE", "GREEN", "RED"])
+          expect(described_class.for(turn_message)).to eq("Three colors guessed at the wrong position")
+        end
+      end
+
+      context "two colors guessed at the wrong position" do
+        it "returns a message for two partial guess colors" do
+          turn = Turn.new(passcode: passcode)
+          turn_message = turn.guess(["BLUE", "RED", "ORANGE", "ORANGE"])
+          expect(described_class.for(turn_message)).to eq("Two colors guessed at the wrong position")
+        end
+      end
+
+      context "one color guessed at the wrong position" do
+        it "returns a message for one partial guess colors" do
+          turn = Turn.new(passcode: passcode)
+          turn_message = turn.guess(["BLUE", "ORANGE", "ORANGE", "ORANGE"])
+          expect(described_class.for(turn_message)).to eq("One color guessed at the wrong position")
+        end
+      end
+    end
+
+    describe "exact and partial position" do
+      context "two exact, two partial match" do
+        it "returns a message for two :exact, two :partial guess colors" do
+          turn = Turn.new(passcode: passcode)
+          turn_message = turn.guess(["RED", "BLUE", "GREEN", "YELLOW"])
+          expect(described_class.for(turn_message)).to eq("Four colors guessed, two at the exact position and two at the wrong position")
+        end
+      end
+
+      context "one exact, one partial match" do
+        it "returns a message for one :exact, one :partial guess colors" do
+          turn = Turn.new(passcode: passcode)
+          turn_message = turn.guess(["RED", "BLUE", "ORANGE", "ORANGE"])
+          expect(described_class.for(turn_message)).to eq("Two colors guessed, one at the exact position and one at the wrong position")
+        end
+      end
+
+      # context "two exact, one partial match" do
```

Comment:
There are still cases that need implementation, please cover all possible scenarios.

Also, consider if the order of result items is important. For example, is `[:exact, :partial]` same or different from `[:partial, :exact]`? If the order doesn't matter, consider sorting the result when it's being returned, so that there's a smaller number of combinations that can be returned.

Reply:
Okay, would surely try it out.